### PR TITLE
Adding components to 1.3.15 and update to hourly build 

### PR DESCRIPTION
**Pull request by jordarlu**
Adding components to 1.3.15 manifest and update the build to be hourly

### Description
1. Adding essential components to 1.3.15 OpenSearch manifest: common-utils + job-scheduler + security
2. Adding all components to 1.3.15 OpenSearch Dashboards manifest
3. Updating 1.3.15 to hourly build for both OpenSearch and OpenSearch Dashboards
4. Updating maven publish 1.3.x to use manifest 1.3.15

### Issues Resolved
part of https://github.com/opensearch-project/opensearch-build/issues/4294

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/1.3.15/opensearch-1.3.15.yml
Comment (zelinh) [inline manifests/1.3.15/opensearch-1.3.15.yml:1]: You could add `depends_on` to plugins and update the schema version to enable incremental build.

**Review by zelinh — COMMENTED** [review]
Please make sure you have all version bumped to 1.3.15 for plugin repo. For some plugins, they need dependencies plugins to be built for their CI to pass.

File: manifests/1.3.15/opensearch-1.3.15.yml
Comment (jordarlu) [reply manifests/1.3.15/opensearch-1.3.15.yml:1]: thank you, @zelinh , let me check those plugins that haven't bump to 1.3.15 yet...

Comment (jordarlu) [reply manifests/1.3.15/opensearch-1.3.15.yml:1]: thanks, @zelinh , I change to have only those essential components for now .. place holder PR after https://github.com/opensearch-project/OpenSearch/issues/12243 is resolved.

Comment (jordarlu) [reply manifests/1.3.15/opensearch-1.3.15.yml:1]: The core build was successful and AUTOCUT issue was resolved. https://build.ci.opensearch.org/job/distribution-build-opensearch/9492/

Comment (jordarlu) [reply manifests/1.3.15/opensearch-1.3.15.yml:1]: https://github.com/opensearch-project/common-utils/pull/575 has been merged

**Review by zelinh — APPROVED** [review]
LGTM! I think we normally added OSD plugins after building their backend plugins but it should be fine. Thanks @jordarlu

**Pull request merged**